src/functions-and-arrays.js

```diff
@@ -1,119 +1,261 @@
+      if (typeof el === "number") {
+        value = el;
+      }
+      if (typeof el === "object" || typeof el === "array") {
+        throw "msg";
+      }
+      acc += value;
+    });
+    return acc;
+  }
+}
+sum([6, 12, "miami", 1, true, "barca", "200", "lisboa", 8, 10]);
 
 // Iteration #4: Calculate the average
 // Level 1: Array of numbers
 const numbersAvg = [2, 6, 9, 10, 7, 4, 1, 9];
 
-function averageNumbers() {}
-
+function averageNumbers(arrayToManipulate) {
+  if (arrayToManipulate.length < 1) return null;
+  let i = 0;
+  arrayToManipulate.forEach((el) => (i += el));
+  return i / arrayToManipulate.length;
+}
 
 // Level 2: Array of strings
-const wordsArr = ['seat', 'correspond', 'linen', 'motif', 'hole', 'smell', 'smart', 'chaos', 'fuel', 'palace'];
+const wordsArr = [
+  "seat",
+  "correspond",
+  "linen",
+  "motif",
+  "hole",
+  "smell",
+  "smart",
+  "chaos",
+  "fuel",
+  "palace",
+];
 
-function averageWordLength() { }
+function averageWordLength(arrayData) {
+  if (arrayData.length < 1) return null;
+
+  let i = 0;
+  arrayData.forEach((el) => (i += el.length));
+  return i / arrayData.length;
+}
 
 // Bonus - Iteration #4.1
-function avg() {}
+function avg(arrayData) {
+  let i = 0;
+  arrayData.forEach((el) => (i += el));
+  return i / arrayData.length;
+}
 
 // Iteration #5: Unique arrays
 const wordsUnique = [
-  'crab',
-  'poison',
-  'contagious',
-  'simple',
-  'bring',
-  'sharp',
-  'playground',
-  'poison',
-  'communion',
-  'simple',
-  'bring'
+  "crab",
+  "poison",
+  "contagious",
+  "simple",
+  "bring",
+  "sharp",
+  "playground",
+  "poison",
+  "communion",
+  "simple",
+  "bring",
 ];
+function uniquifyArray(arrayData) {
+  if (arrayData.length === 0) return null;
+  let arrayToTest = arrayData.filter(
+    (el, index) => index === arrayData.indexOf(el)
+  );
+  return arrayToTest;
+}
 
-function uniquifyArray() {}
-
-
+uniquifyArray(wordsUnique);
 
 // Iteration #6: Find elements
-const wordsFind = ['machine', 'subset', 'trouble', 'starting', 'matter', 'eating', 'truth', 'disobedience'];
-
-function doesWordExist() {}
-
+const wordsFind = [
+  "machine",
+  "subset",
+  "trouble",
+  "starting",
+  "matter",
+  "eating",
+  "truth",
+  "disobedience",
+];
 
+function doesWordExist(arrayData, word) {
+  if (arrayData.length === 0) return null;
+  let booleanToReturn = false;
+  arrayData.forEach((el) => (word === el ? (booleanToReturn = true) : ""));
+  return booleanToReturn;
+}
+doesWordExist(wordsFind, "machin");
 
 // Iteration #7: Count repetition
 const wordsCount = [
-  'machine',
-  'matter',
-  'subset',
-  'trouble',
-  'starting',
-  'matter',
-  'eating',
-  'matter',
-  'truth',
-  'disobedience',
-  'matter'
+  "machine",
+  "matter",
+  "subset",
+  "trouble",
+  "starting",
+  "matter",
+  "eating",
+  "matter",
+  "truth",
+  "disobedience",
+  "matter",
+  "matter",
+  "matter",
 ];
 
-function howManyTimes() {}
-
-
+function howManyTimes(arrayData, wordToSearch) {
```

Comment:
This function is not working if the word repeats more than one time. A way of doing this is by creating a `loop` that will check if the specific `word` exists in the `array` and if it does `.push()` it inside a new `array`:

```
function howManyTimes(array, word) {
  if (array.length === 0){
    return 0;
  }
  let newArray = [];
  for (let i=0; i<array.length; i++){
    if (array[i] === word){
      newArray.push(array[i])
    }
  }
  return newArray.length
}
```

Reply:
man, try this code:
`const wordsCount = [
  "machine",
  "matter",
  "subset",
  "trouble",
  "starting",
  "matter",
  "eating",
  "matter",
  "truth",
  "disobedience",
  "matter",
  "matter",
  "matter",
];

function howManyTimes(arrayData, wordToSearch) {
  if (arrayData.length === 0) return 0;
  if (arrayData.indexOf(wordToSearch) === -1) return 0;
  let tst = arrayData.filter((el, index) =>
    el === wordToSearch ? arrayData.indexOf(wordToSearch) !== index : ""
  );
  if (tst.length > 0) {
    console.log(tst.length);
    return tst.length;
  }
  if (tst.length === 0) return 1;
}
howManyTimes(wordsCount, "matter");`
I think that works when the word repeats more than one time!